### PR TITLE
mitosheet: add reset index step

### PR DESCRIPTION
**Pull request by naterush**
# Description

Pretty simple. Seems incredibly useful given the formulas, so I went and implemented!

# Testing

See tests. Maybe add some.

# Documentation

We'll need to write notes!

### CODE REVIEW

File: mitosheet/mitosheet/code_chunks/reset_index_code_chunk.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        self.sheet_index = sheet_index
+        self.drop = drop
+
+        self.df_name = self.post_state.df_names[self.sheet_index]
+
+    def get_display_name(self) -> str:
+        return 'Reset Index'
+    
+    def get_description_comment(self) -> str:
+        return "Reset dataframe index"
```
Comment (naterush) [inline mitosheet/mitosheet/code_chunks/reset_index_code_chunk.py:24]: Update comment

File: mitosheet/mitosheet/tests/test_utils.py [hunk 2]
```diff
@@ -512,6 +512,30 @@ def excel_range_import(
...
     
+
+    @check_transpiled_code_after_call
+    def reset_index(
+            self, 
+            sheet_index: int,
+            drop: bool,
+        ) -> bool:
+
+
```
Comment (naterush) [inline mitosheet/mitosheet/tests/test_utils.py:523]: Cleanup

**Review by naterush — COMMENTED** [review]
Cleanup!

File: mitosheet/src/utils/actions.tsx [hunk 3]
```diff
@@ -1281,11 +1281,34 @@ export const createActions = (
...
+        [ActionEnum.RESET_AND_KEEP_INDEX]: {
+            type: ActionEnum.RESET_AND_KEEP_INDEX,
+            shortTitle: 'Reset Index',
+            longTitle: 'Reset Index',
+            actionFunction: () => {
+                void mitoAPI.editResetIndex(sheetIndex, false);
+            },
+            isDisabled: () => {return doesAnySheetExist(sheetDataArray) ? undefined : 'Import data before resetting an index.'},
+            searchTerms: ['reset', 'index'],
+            tooltip: "Resets a dataframes index to 0-NUM_ROWS. Will keep the current index as a column in the dataframe."
```
Comment (aarondr77) [inline mitosheet/src/utils/actions.tsx:1293]: 0-NUM_ROWS was confusing to me. Maybe we could do `0,1,2...` 

Update in the next `ActionEnum.RESET_AND_DROP_INDEX` action too

File: mitosheet/src/utils/actions.tsx [hunk 4]
```diff
@@ -1281,11 +1281,34 @@ export const createActions = (
...
+        },
+        [ActionEnum.RESET_AND_DROP_INDEX]: {
+            type: ActionEnum.RESET_AND_DROP_INDEX,
+            shortTitle: 'Reset and Drop Index',
+            longTitle: 'Reset and Drop Index',
+            actionFunction: () => {
+                void mitoAPI.editResetIndex(sheetIndex, true);
+            },
+            isDisabled: () => {return doesAnySheetExist(sheetDataArray) ? undefined : 'Import data before resetting an index.'},
+            searchTerms: ['Reset Index'],
```
Comment (aarondr77) [inline mitosheet/src/utils/actions.tsx:1303]: I think we want our search terms to be lowercase. And to keep it consistent, with the above should be: ['reset', 'index']

**Comment by aarondr77** [comment]
LGTM after release

**Pull request merged**